### PR TITLE
Enhance sync

### PR DESCRIPTION
**Pull request by mynameisyz**
To do:
- [x] handle sync exceptions
- [x] sync test case

Known issues:
- Opus hangs when the user closes the browser link without allowing/denying permission. See [link](http://stackoverflow.com/questions/30118487/how-to-get-result-from-google-oauth-api-if-user-close-browser-window/43141043#43141043)

Bonus:
- [ ] logout

### CODE REVIEW

File: src/main/java/seedu/opus/sync/SyncServiceGtask.java [hunk 1]
```diff
@@ -228,11 +249,27 @@ private TaskList createOpusTaskList() throws SyncException {
...
-                                    : TASK_STATUS_INCOMPLETE);
+                ? TASK_STATUS_COMPLETE
+                        : TASK_STATUS_INCOMPLETE);
         return googleAdaptedTask;
     }
 
+    /**
+     * Deauthorize Opus by deleting the storedCredential file
+     */
+    public void deauthorize() {
```
Comment (xbili) [inline src/main/java/seedu/opus/sync/SyncServiceGtask.java:260]: I think `unauthorize` sounds more correct, it's a small matter. 😅

File: src/test/java/seedu/opus/logic/SyncCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package seedu.opus.logic;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.doThrow;
```
Comment (xbili) [inline src/test/java/seedu/opus/logic/SyncCommandTest.java:5]: 👏

**Review by xbili — APPROVED** [review]
LGTM! 👍  Mockito yayyyyy

File: src/main/java/seedu/opus/sync/SyncServiceGtask.java (hunk 1 above)
Comment (mynameisyz) [reply src/main/java/seedu/opus/sync/SyncServiceGtask.java:260]: Well I though that `deauthorize` is more of a verb while `unauthorize` is more of an adjective.

**Pull request merged**